### PR TITLE
Widget Support revisions

### PR DESCRIPTION
**Pull request by kshala-ford**
This PR contains revisions to Widget Support being under review from May 8.

### CODE REVIEW

File: proposals/0216-widget-support.md [hunk 1]
```diff
@@ -243,6 +259,23 @@ The enum `PredefinedWindows` specifies what windows and IDs are predefined and p
...
 
+The primary widget is a special widget, that can be associated with a service type, which is used by the HMI whenever a single widget needs to represent the whole app. The primary widget should be named as the app and can be pre-created by the HMI.
+
+#### Associated Service type
+
+This section describes a possible feature to allow widgets per app or service type. Instead of providing only app specific widgets, an infotainment system with widget support can also offer widgets per type. As an example, the infotainment system can offer a "Media Player" widget which will present the widget of the active/audible media application. This can include media sources offered by the infotainment system such as Radio, Bluetooth or USB playback. A similar user experience can be offered to other app types like navigation, weather, (voice) communication or telephony, voice assistant etc. 
+
+`assiciatedServiceType` allows an app to create a widget related to a specific service type.
```
Comment (theresalech) [inline proposals/0216-widget-support.md]:
```suggestion
`associatedServiceType` allows an app to create a widget related to a specific service type.
```

File: proposals/0216-widget-support.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 * Proposal: [SDL- 0216](0216-widget-support.md)
 * Author: [Ashwin Karemore](https://github.com/ashwink11), [Kujtim Shala](https://github.com/kshala-ford)
-* Status: **In Review**
+* Status: **Returned for Revisions**
```
Comment (theresalech) [inline proposals/0216-widget-support.md]:
```suggestion
* Status: **Accepted with Revisions**
```

based on the decision made in the 2019-05-14 Steering Committee meeting.

**Comment by AByzhynar** [comment]
@kshala-ford @kshala-ford Please return back removed parameters to `SetDisplayLayout` RPC to guarantee compatibility with old applications. 
They were removed here: 
https://github.com/smartdevicelink/sdl_evolution/pull/729/files#diff-ba08fc82743036684a177b4b4c32eda4L337

**Comment by joeljfischer** [comment]
@AByzhynar I don't see any removed parameters from `SetDisplayLayout`. Can you specify what you mean?

**Comment by AByzhynar** [comment]
@joeljfischer Please see how `SetDisplayLayout` look in MobileAPI.xml on release 5.1.1: 

```        
    <function name="SetDisplayLayout" functionID="SetDisplayLayoutID" messagetype="request" since="3.0">
        <description>
            Used to set an alternate display layout.
            If not sent, default screen for given platform will be shown
        </description>
        
        <param name="displayLayout" type="String" maxlength="500" mandatory="true">
            <description>
                Predefined or dynamically created screen layout.
                Currently only predefined screen layouts are defined.
            </description>
        </param>

        <param name="dayColorScheme" type="TemplateColorScheme" mandatory="false" since="5.0"/>

        <param name="nightColorScheme" type="TemplateColorScheme" mandatory="false" since="5.0"/>
    </function>
    
    <function name="SetDisplayLayout" functionID="SetDisplayLayoutID" messagetype="response" since="3.0">
        
        <param name="success" type="Boolean" platform="documentation" mandatory="true">
            <description> true, if successful; false, if failed </description>
        </param>
        
        <param name="resultCode" type="Result" platform="documentation" mandatory="true">
            <description>See Result</description>
            <element name="SUCCESS"/>
            <element name="INVALID_DATA"/>
            <element name="OUT_OF_MEMORY"/>
            <element name="TOO_MANY_PENDING_REQUESTS"/>
            <element name="APPLICATION_NOT_REGISTERED"/>
            <element name="GENERIC_ERROR"/>
            <element name="REJECTED"/>
            <element name="UNSUPPORTED_REQUEST"/>
        </param>
        
        <param name="displayCapabilities" type="DisplayCapabilities" mandatory="false">
            <description>See DisplayCapabilities</description>
        </param>
        
        <param name="buttonCapabilities" type="ButtonCapabilities" minsize="1" maxsize="100" array="true" mandatory="false">
            <description>See ButtonCapabilities</description >
        </param>
        
        <param name="softButtonCapabilities" type="SoftButtonCapabilities" minsize="1" maxsize="100" array="true" mandatory="false">
            <description>If returned, the platform supports on-screen SoftButtons; see SoftButtonCapabilities.</description>
        </param>
        
        <param name="presetBankCapabilities" type="PresetBankCapabilities" mandatory="false">
            <description>If returned, the platform supports custom on-screen Presets; see PresetBankCapabilities.</description>
        </param>
        
        <param name="info" type="String" maxlength="1000" mandatory="false" platform="documentation">
            <description>Provides additional human readable info regarding the result.</description>
        </param>
        
    </function>
```

and in proposal `SetDisplayLayout` request/response were marked as `deprecated` which is good, but at the same time all parameters from request/response were removed. 
As new SetDisplayLayout RPCs in proposal are shown as :  
```
<function name="SetDisplayLayout" functionID="SetDisplayLayoutID" messagetype="request" deprecated="true" since="5.x">
    <description>This RPC is deprecated. Use Show RPC to change layout.</description>
</function>

<function name="SetDisplayLayout" functionID="SetDisplayLayoutID" messagetype="response" deprecated="true" since="5.x">
    <description>This RPC is deprecated. Use Show RPC to change layout.</description>
</function>
```
Looks that old parameters should be kept as well as `deprecated` mark for RPC.

**Comment by joeljfischer** [comment]
@AByzhynar He just wasn't including the parameters since they didn't change and it would just clutter the document. He should have put a comment mentioning that the parameters stay there. Because the entire RPC is deprecated, the parameters don't need to be marked, I believe.

**Comment by AByzhynar** [comment]
> @AByzhynar He just wasn't including the parameters since they didn't change and it would just clutter the document. He should have put a comment mentioning that the parameters stay there. Because the entire RPC is deprecated, the parameters don't need to be marked, I believe.

I think so. And I agree with you that it should be explicitly stated.

File: proposals/0216-widget-support.md [hunk 3]
```diff
@@ -351,6 +384,18 @@ The `Show` request will take over the responsibilities of `SetDisplayLayout` by
...
 ```xml
+<struct name="TemplateConfiguration" since="5.x">
+  <param name="template" type="String" maxlength="500" mandatory="true">
+    <description>
+        Predefined or dynamically created window template.
+        Currently only predefined window template layouts are defined.
+    </description>
+  </param>
+
+  <param name="dayColorScheme" type="TemplateColorScheme" mandatory="false" />
```
Comment (AByzhynar) [inline proposals/0216-widget-support.md]: @kshala-ford Can you specify definition of type `TemplateColorScheme`. I can find it nowhere. Thank you

**Comment by kshala-ford** [comment]
@theresalech I have made the requested changes. The other questions seems to be answered already.

**Pull request merged**